avro-fastserde/src/codegen/java/com/linkedin/avro/fastserde/generated/deserialization/AVRO_1_4/Array_of_UNION_GenericDeserializer_585074122056792963_585074122056792963.java

```diff
@@ -44,12 +44,15 @@ public Array_of_UNION_GenericDeserializer_585074122056792963_585074122056792963(
                     arrayArrayElementReuseVar0 = ((GenericArray)(reuse)).peek();
                 }
                 int unionIndex0 = (decoder.readIndex());
-                if (unionIndex0 == 0) {
-                    decoder.readNull();
-                } else {
-                    if (unionIndex0 == 1) {
+                switch (unionIndex0) {
+                    case  0 :
+                        decoder.readNull();
+                        break;
+                    case  1 :
                         array0 .add(deserializerecord0(arrayArrayElementReuseVar0, (decoder)));
-                    }
+                        break;
+                    default:
+                        throw new RuntimeException(("Illegal union index: "+ unionIndex0));
```

Comment:
this exception can be somewhat more useful - print the field name? does decoder have a curreptPosition you could print?

Reply:
Potentially difficult, but I'll try.

Reply:
Done.